orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/gce/GoogleServerGroupCreator.groovy

```diff
@@ -49,34 +50,39 @@ class GoogleServerGroupCreator implements ServerGroupCreator, DeploymentDetailsA
       operation.credentials = operation.account
     }
 
-    operation.image = operation.image ?: getImage(stage)
-
-    if (!operation.image) {
-      throw new IllegalStateException("No image could be found in ${stage.context.region}.")
+    if (stage.context.imageSource == "artifact") {
+      operation.imageSource = "ARTIFACT"
+      operation.imageArtifact = getImageArtifact(stage)
+    } else {
+      operation.imageSource = "STRING"
+      operation.image = operation.image ?: getImage(stage)
+      if (!operation.image) {
+        throw new IllegalStateException("No image could be found in ${stage.context.region}.")
+      }
     }
 
     return [[(ServerGroupCreator.OPERATION): operation]]
   }
 
-  private String getImage(Stage stage) {
+  private Artifact getImageArtifact(Stage stage) {
     def stageContext = stage.getContext()
+
+    def artifactId = stageContext.imageArtifactId as String
+    if (artifactId == null) {
+      throw new IllegalStateException("Image source was set to artifact but no artifact was specified.")
```

Comment:
I think `IllegalArgumentException` is a little more 'correct'

Reply:
Makes sense, I'll change this